tensorflow/python/training/optimizer.py

```diff
@@ -27,7 +27,7 @@
 from tensorflow.python.ops import state_ops
 from tensorflow.python.ops import variables
 from tensorflow.python.training import slot_creator
-
+import tensorflow as tf
```

Comment:
Remove this import (as it creates a circular dependency, and the relevant method is available from `ops`).